src/briefcase/integrations/android_sdk.py

```diff
@@ -212,61 +248,305 @@ def verify_emulator(self):
+                else:
+                    # Specifier is a direct device ID
+                    avd = None
+                    device = device_or_avd
+
+                details = running_devices[device]
+                avd = details.get("avd")
+                if details["authorized"]:
+                    # An authorized, running device (emulator or physical)
+                    return device, name, avd
+                else:
+                    # An unauthorized physical device
+                    raise AndroidDeviceNotAuthorized(device)
+
+            except KeyError:
+                # Provided device_or_id isn't a valid device identifier.
+                if device_or_avd.startswith("@"):
+                    id_type = "emulator AVD"
+                else:
+                    id_type = "device ID"
+                raise InvalidDeviceError(id_type, device_or_avd)
+
+        # We weren't given a device/AVD; we have to select from the list.
+        # If we're selecting from a list, there's always one last choice
+        choices.append((None, "Create a new Android emulator"))
+
+        # Show the choices to the user.
+        print()
+        print("Select device:")
+        print()
+        choice = select_option(choices, input=self.command.input)
+
+        # Proces the user's choice
+        if choice is None:
+            # Create a new emulator. No device ID or AVD.
+            device = None
+            avd = None
+            name = None
+        elif choice.startswith("@"):
+            # A non-running emulator. We have an AVD, but no device ID.
+            device = None
+            name = device_choices[choice]
+            avd = choice[1:]
+        else:
+            # Either a running emulator, or a physical device. Regardless,
+            # we need to check if the device is developer enabled
+            try:
+                details = running_devices[choice]
+                if not details["authorized"]:
+                    # An unauthorized physical device
+                    raise AndroidDeviceNotAuthorized(choice)
+
+                # Return the device ID and name.
+                device = choice
+                name = device_choices[choice]
+                avd = details.get("avd")
+            except KeyError:
+                raise InvalidDeviceError("device ID", choice)
+
+        if avd:
+            print()
+            print("In future, you could specify this device by running:")
+            print()
+            print("    briefcase run android -d @{avd}".format(avd=avd))
+        elif device:
+            print()
+            print("In future, you could specify this device by running:")
+            print()
+            print("    briefcase run android -d {device}".format(device=device))
+
+        return device, name, avd
+
+    def create_emulator(self):
+        """Create a new Android emulator.
+
+        """
+        print()
+        name = self.command.input("Emulator name: ")
```

Comment:
I would personally suggest we always use the name "briefcasephone" or something, or at least that we provide that as a default. I believe many people will be surprised they have to choose a name here.

Reply:
Ah - but the eventual plan is that there will be the ability to select different device profiles, sizes, etc (There's a TODO in the branch to create the actual devices).